pkg/check/node_esxi_version.go

```diff
@@ -0,0 +1,86 @@
+package check
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/vmware/govmomi/object"
+	"github.com/vmware/govmomi/vim25/mo"
+	v1 "k8s.io/api/core/v1"
+	"k8s.io/component-base/metrics"
+	"k8s.io/component-base/metrics/legacyregistry"
+	"k8s.io/klog/v2"
+)
+
+// CollectNodeESXiVersion emits metric with version of each ESXi host that runs at least a single VM with node.
+type CollectNodeESXiVersion struct {
+	// map ESXI host name ("host-12345", not hostname nor IP address!) -> version
+	esxiVersions map[string]string
+}
+
+var _ NodeCheck = &CollectNodeESXiVersion{}
+
+var (
+	esxiVersionMetric = metrics.NewGaugeVec(
+		&metrics.GaugeOpts{
+			Name:           "vsphere_esxi_version",
```

Comment:
Reading https://github.com/kubernetes/community/blob/master/contributors/devel/sig-instrumentation/instrumentation.md, should it be `vsphere_esxi_versions_total`?

Reply:
I think we should leave `_total` suffix for accumulating counters.

Reply:
renamed